src/transformers/trainer_seq2seq.py

```diff
@@ -182,23 +182,11 @@ def prediction_step(
             gen_kwargs["synced_gpus"] if gen_kwargs.get("synced_gpus") is not None else default_synced_gpus
         )
 
-        if "attention_mask" in inputs:
-            gen_kwargs["attention_mask"] = inputs.get("attention_mask", None)
-        if "global_attention_mask" in inputs:
-            gen_kwargs["global_attention_mask"] = inputs.get("global_attention_mask", None)
-
-        # prepare generation inputs
-        # some encoder-decoder models can have varying encoder's and thus
-        # varying model input names
-        if hasattr(self.model, "encoder") and self.model.encoder.main_input_name != self.model.main_input_name:
-            generation_inputs = inputs[self.model.encoder.main_input_name]
-        else:
-            generation_inputs = inputs[self.model.main_input_name]
+        # TODO (Joao): the following line is needed to keep a consistent result on SQUAD. Ideally, we should not block
+        # users from preparing a dataset with `decoder_input_ids`.
+        inputs = {k: v for k, v in inputs.items() if k != "decoder_input_ids"}
```

Comment:
Why did you have to add this? It's probably something badly coded in the run_seq2seq_qa example.

Reply:
@sgugger 
It's returned as part of the DataCollator for Seq2Seq -- see [here](https://github.com/huggingface/transformers/blob/main/src/transformers/data/data_collator.py#L600). This means that, [for T5](https://github.com/huggingface/transformers/blob/main/src/transformers/models/t5/modeling_t5.py#L1783), `decoder_input_ids` = labels shifted right. This is needed at train time, for teacher forcing.

However, when we pass `decoder_input_ids` to generate, it continues generating from those tokens. This means that we are, in essence, continuing generation after the labels, which doesn't make sense for evaluation purposes. 

When writing this comment, I noticed that our models have [this line](https://github.com/huggingface/transformers/blob/987972377d8336a9a17906d8b2c9a650f7ad04aa/src/transformers/models/t5/modeling_t5.py#L1688), which create `decoder_input_ids` when `labels` is passed. This means that we can probably remove `decoder_input_ids` from the DataCollator output (under a flag, to avoid backward compatibility problems), as well as this line here. WDYT?

Reply:
No, we need to pass them as the decoder input IDs generated by the model do not take the padding into account IIRC, hence the line in the data collator. This is all to properly support label smoothing.

Reply:
😢